platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/SOCHomePageTestIT.java

```diff
@@ -0,0 +1,885 @@
+    activityStream.deleteactivity(text1);
+    homePagePlatform.goToMySpaces();
+    spaceManagement.deleteSpace(space, false);
+    manageLogInOut.signIn(DATA_USER2, DATA_PASS);
+    activityStream.deleteactivity(text);
+    homePagePlatform.goToConnections();
+    connectionsManagement.removeConnection(DATA_USER1);
+
+  }
+
+  /**
+   * <li>Case ID:121932.</li>
+   * <li>Test Case Name: Check activities order.</li>
+   * <li>Pre-Condition:</li>
+   * <li>Post-Condition:</li> Step Number: 1 Step Name: - Check order of
+   * activities Step Description: - Log in and go to homepage - Check order of
+   * activity Input Data: Expected Outcome: - Homepage is displayed - the order of
+   * activities is based on activity's last action date. The last action date is
+   * the latest of: 1. The publication date 2. The date of the last comment posted
+   */
+
+  @Test
+  public void test14_CheckActivitiesOrder() {
+    info("Test 14 Check activities order");
+    String text = "text" + getRandomNumber();
+    String text1 = "text1" + getRandomNumber();
+    String comment = "comment" + getRandomNumber();
+    activityStream.addActivity(text, "");
+    refresh();
+    activityStream.addActivity(text1, "");
+    ELEMENT_ACTIVITY_STREAM_CONTAINER.findAll(ELEMENT_BOX_ACTIVITY).get(0).find(byText(text1)).should(Condition.exist);
+    ELEMENT_ACTIVITY_STREAM_CONTAINER.findAll(ELEMENT_BOX_ACTIVITY).get(1).find(byText(text)).should(Condition.exist);
+    activityStream.commentActivity(text, comment);
+    refresh();
+    ELEMENT_ACTIVITY_STREAM_CONTAINER.findAll(ELEMENT_BOX_ACTIVITY).get(0).find(byText(text)).should(Condition.exist);
+    executeJavaScript("window.scrollBy(0,-1500)", "");
+    activityStream.deleteactivity(text);
+    executeJavaScript("window.scrollBy(0,-1500)", "");
+    activityStream.deleteactivity(text1);
+  }
+
+  /**
+   * <li>Case ID:121934.</li>
+   * <li>Test Case Name: Check Layout of Activities.</li>
+   * <li>Pre-Condition:</li>
+   * <li>Post-Condition:</li> Step Number: 1 Step Name: Check layout of activity
+   * Step Description: - log in and goto intranet hompage - Select an activity to
+   * check its layout Input Data: Expected Outcome: An activity is made out of
+   * different parts:(see attached) 1.the author 2.the author's avatar 3.the space
+   * (optional) 4.the type (optional) 5.the activity message 6.the featured
+   * content (optional) 7.the Action bar (Comment and Like links + custom actions)
+   * 8.the like section (optional) 9.the comment section (optional)
+   */
+  @Test
+  public void test16_CheckLayoutOfActivities() {
+    info("Test 16 Check Layout of Activities");
+
+    String name = "" + getRandomNumber();
+
+    homePagePlatform.goToHomePage();
+    activityStream.addActivity(name, "");
+    $(byText(name)).parent()
+                   .parent()
+                   .parent()
+                   .parent()
+                   .parent()
+                   .parent()
+                   .parent()
```

Comment:
This looks very fragile. No way to use an id or a class somewhere in the DOM ?

Reply:
$(byText(name)).parent() .parent() .parent() .parent().parent().parent().parent().find(ELEMENT_PUBLICATION_FIRSTPOST_AUTHORAVATAR).should(Condition.exist);